flink-python/pyflink/streaming/stream_execution_environment.py

```diff
@@ -0,0 +1,413 @@
+        :param serializer_class_name: The full-qualified java class name of the serializer to use.
+        """
+        type_clz = load_java_class(type_class_name)
+        j_serializer_clz = load_java_class(serializer_class_name)
+        self._j_stream_execution_environment.registerTypeWithKryoSerializer(
+            type_clz, j_serializer_clz)
+
+    def register_type(self, type_class_name):
+        """
+        Registers the given type with the serialization stack. If the type is eventually
+        serialized as a POJO, then the type is registered with the POJO serializer. If the
+        type ends up being serialized with Kryo, then it will be registered at Kryo to make
+        sure that only tags are written.
+
+        :param type_class_name: The full-qualified java class name of the type to register.
+        """
+        type_clz = load_java_class(type_class_name)
+        self._j_stream_execution_environment.registerType(type_clz)
+
+    def set_stream_time_characteristic(self, characteristic):
+        """
+        Sets the time characteristic for all streams create from this environment, e.g., processing
+        time, event time, or ingestion time.
+
+        If you set the characteristic to IngestionTime of EventTime this will set a default
+        watermark update interval of 200 ms. If this is not applicable for your application
+        you should change it using
+        :func:`pyflink.common.ExecutionConfig.set_auto_watermark_interval`.
+
+        :param characteristic: The time characteristic.
+        """
+        gateway = get_gateway()
+        JTimeCharacteristic = gateway.jvm.org.apache.flink.streaming.api.TimeCharacteristic
+        if characteristic == TimeCharacteristic.EventTime:
+            j_characteristic = JTimeCharacteristic.EventTime
+        elif characteristic == TimeCharacteristic.IngestionTime:
+            j_characteristic = JTimeCharacteristic.IngestionTime
+        elif characteristic == TimeCharacteristic.ProcessingTime:
+            j_characteristic = JTimeCharacteristic.ProcessingTime
+        else:
+            raise TypeError("Unsupported time characteristic: %s, supported time characteristic "
+                            "are: TimeCharacteristic.EventTime, TimeCharacteristic.IngestionTime, "
+                            "TimeCharacteristic.ProcessingTime." % characteristic)
+        self._j_stream_execution_environment.setStreamTimeCharacteristic(j_characteristic)
+
+    def get_stream_time_characteristic(self):
+        """
+        Gets the time characteristic.
+
+        see :func:`set_stream_time_characteristic`
+
+        :return: The :class:`TimeCharacteristic`.
+        """
+        gateway = get_gateway()
+        JTimeCharacteristic = gateway.jvm.org.apache.flink.streaming.api.TimeCharacteristic
+        j_characteristic = self._j_stream_execution_environment.getStreamTimeCharacteristic()
+        if j_characteristic == JTimeCharacteristic.EventTime:
```

Comment:
The same code logic exists in both set_stream_time_characteristic and get_stream_time_characteristic. What about abstract them in a static method in TimeCharacteristic? Should also consider other places such as set_execution_mode/get_execution_mode, etc.